### ISSUE TITLE
Mautic/core-lib requires non-existst dependency

### ISSUE THREAD

**Issue by mmarton** [mautic/mautic#10651]
### Mautic Version

4.0.0

### PHP version

7.4.26

### What happened?

Hi!

I'm experimenting with plugin development and tried to require the core-bundle to be able to run phpunit phpstan and others.

Problem 1
    - mautic/core-lib[4.0.0-alpha1, ..., 4.x-dev] require friendsofsymfony/oauth-server-bundle dev-doctrine-fix -> found friendsofsymfony/oauth-server-bundle[dev-master, 1.0.0, ..., 1.7.x-dev, 2.0.0-alpha.0, 2.0.x-dev (alias of dev-master)] but it does not match the constraint.
    - Root composer.json requires mautic/core-lib ^4.0 -> satisfiable by mautic/core-lib[4.0.0-alpha1, ..., 4.x-dev].

both code-lib v4 and v3 depends on a non existent branch.
for 4 it is friendsofsymfony/oauth-server-bundle dev-doctrine-fix (https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/branches)
for 3 it is  willdurand/oauth-server-bundle dev-release-0.0.3 (https://github.com/willdurand/BazingaOAuthServerBundle/branches)

### How can we reproduce this issue?

composer req mautic/core-lib

### Relevant log output

```shell
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - mautic/core-lib[4.0.0-alpha1, ..., 4.x-dev] require friendsofsymfony/oauth-server-bundle dev-doctrine-fix -> found friendsofsymfony/oauth-server-bundle[dev-master, 1.0.0, ..., 1.7.x-dev, 2.0.0-alpha.0, 2.0.x-dev (alias of dev-master)] but it does not match the constraint.
    - Root composer.json requires mautic/core-lib ^4.0 -> satisfiable by mautic/core-lib[4.0.0-alpha1, ..., 4.x-dev].
```
```


### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by omert37** [comment]
Same problem here.
Trying to install to new directory with COMPOSER and I get the same message.

 **Problem 1
    - friendsofsymfony/oauth-server-bundle dev-doctrine-fix requires php ^5.5|^7.0 -> your php version (8.0.21) does not satisfy that requirement.
    - mautic/core-lib 4.4.0 requires friendsofsymfony/oauth-server-bundle dev-doctrine-fix -> satisfiable by friendsofsymfony/oauth-server-bundle[dev-doctrine-fix].
    - Root composer.json requires mautic/core-lib 4.4.0 -> satisfiable by mautic/core-lib[4.4.0].**

**Comment by escopecz** [comment]
Please test https://github.com/mautic/mautic/pull/11353 today so it can be released in 4.4.1 that should go out today

**Comment by omert37** [comment]
Tested it out.
It didn't give me the "Problem" errors but it gave me some notices in yellow about bundles being abandoned and that I shouldn't use them.

**Comment by escopecz** [comment]
@omert37 thanks for testing. Those errors are there without this PR as well. Those will be resolved with Mautic 5 as replacement of many of those deprecated packages would be a breaking compatibility change.

**Comment by escopecz** [comment]
@omert37 can you please approve the PR? That way it will be one step closer to being merged.

**Comment by omert37** [comment]
I do not feel I'm in a position to approve or disapprove of anything. I am merely an observer of a process.

**Comment by RCheesley** [comment]
@omert37 the way GitHub knows if people have tested is leaving a review, with 'approve' if the test was successful. We need two to merge every PR, so your approval does matter! You can do that under Files Changed > Leave review 😄

**Comment by omert37** [comment]
Done. I hope it helps.

**Comment by escopecz** [comment]
I made a mistake. The fix I did will break Mautic's API because the master branch does not support PHP templates but only TWIG. We'll have to create a new hot fix reverting the change I made. Now I came back to this issue again and thinking why we are getting the error you've described. 

I think I start to understand it. Let's get back to the roots. Do not use composer to install Mautic in your plugin repository. In other words, do not run `composer req mautic/core-lib` in your plugin repository. Just paste `"mautic/core-lib": "^4.3"` to your `require` section in your composer.json.

It is there just to so Composer could throw an error if anyone would try to install it to a Mautic version that does not support your plugin.

Does it make sense?
